### PR TITLE
net: openthread: Select all required nrf_security options

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
After disable by default nrf_security options we need to explicitly enable required onces.

- [ ] blocked by https://github.com/nrfconnect/sdk-nrfxlib/pull/288

### CODE REVIEW

File: subsys/net/openthread/Kconfig [hunk 1]
```diff
@@ -21,9 +21,15 @@ config OPENTHREAD_NRF_SECURITY
 	select OBERON_BACKEND if !CC310_BACKEND
 	select MBEDTLS_TLS_LIBRARY
```
Comment (edmont) [inline subsys/net/openthread/Kconfig:22]: Consider alphabetical ordering for the MBEDTLS options here.

**Review by edmont — APPROVED** [review]

**Review by MarekPorwisz — APPROVED** [review]

**Comment by edmont** [comment]
Updated since new recently merged Oberon now supports SHA224: https://github.com/nrfconnect/sdk-nrfxlib/commit/b6dfe1eaa91fbb897edabf4d8b7f747d6f46a05c

@frkv please confirm.

**Comment by frkv** [comment]
> Updated since new recently merged Oberon now supports SHA224: [nrfconnect/sdk-nrfxlib@b6dfe1e](https://github.com/nrfconnect/sdk-nrfxlib/commit/b6dfe1eaa91fbb897edabf4d8b7f747d6f46a05c)
> 
> @frkv please confirm.

Yes. the nrf_oberon 3.0.7 supports SHA-224, but our CI doesn't have a test for it.

**Comment by lmaciejonczyk** [comment]
Most of changes has been transferred here: https://github.com/nrfconnect/sdk-nrf/pull/2974

**Pull request closed without merging**